lib/services/s3.js

```diff
@@ -479,6 +499,50 @@ AWS.util.update(AWS.S3.prototype, {
     }
   },
 
+  /**
+   * @api private
+   */
+  validateMd5TrailingChecksum: function validateMd5TrailingChecksum(resp) {
```

Comment:
I know it's still a WIP, just noting you won't need this method anymore since you've moved the check closer to the IO!